src/comm/TcpAcceptor.cc

```diff
@@ -265,13 +236,14 @@ Comm::TcpAcceptor::logAcceptError(const ConnectionPointer &tcpClient) const
 }
 
 void
-Comm::TcpAcceptor::acceptOne()
+Comm::TcpAcceptor::acceptOne(const CommIoCbParams &)
 {
-    /*
-     * We don't worry about running low on FDs here.  Instead,
-     * doAccept() will use AcceptLimiter if we reach the limit
-     * there.
-     */
+    debugs(5, 2, "new connection on " << conn);
+
+    if (!okToAccept()) {
+        AcceptLimiter::Instance().defer(this);
+        return;
+    }
```

Comment:
You requested that this class be made to use the Comm::Read API instead of scheduling accept handlers on the old SetSelect() API. On listening sockets a read I/O signal is the trigger that an accept(2) needs to be done.

The acceptOne() logic is what this class Comm::Read handler needs to be doing, with a few modifications for the API differences. It could be restructured internally to looks like other Comm::Read handlers, I left that too make the PR logic changes clearer but its easily in scope.

Reply:
ERR_CLOSING is added by c92b96b.